src/Plugin/EntityBrowser/SelectionDisplay/NewDisplay.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+/**
+ * Contains \Drupal\entity_browser\Plugin\EntityBrowser\SelectionDisplay\NewDisplay.
+ */
+
+namespace Drupal\entity_browser\Plugin\EntityBrowser\SelectionDisplay;
+
+use Drupal\Component\Utility\Html;
+use Drupal\Component\Utility\NestedArray;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\entity_browser\SelectionDisplayBase;
+
+/**
+ * Show current selection and delivers selected entities.
+ *
+ * @EntityBrowserSelectionDisplay(
+ *   id = "new_display",
+ *   label = @Translation("New selection display"),
+ *   description = @Translation("Show current selection display and delivers selected entities.")
+ * )
+ */
+class NewDisplay extends SelectionDisplayBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    return array(
+      'view' => NULL,
+      'view_display' => NULL,
+    ) + parent::defaultConfiguration();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getForm(array &$original_form, FormStateInterface $form_state) {
+    $form = [];
+    $form['#attached']['library'][] = 'entity_browser/new_display';
+
+    $selected_entities = $form_state->get(['entity_browser', 'selected_entities']);
+    $ids = array_keys($selected_entities);
+
+    $hidden_id = Html::getUniqueId('edit-selected-target-id');
+    $details_id = Html::getUniqueId('edit-selected');
+
+    $form += [
+      '#id' => $details_id,
+      '#type' => 'details',
+      '#open' => !empty($ids),
+      'target_id' => [
+        '#type' => 'hidden',
+        '#id' => $hidden_id,
+        // We need to repeat ID here as it is otherwise skipped when rendering.
+        '#attributes' => ['id' => $hidden_id],
+        '#default_value' => $ids,
+        // #ajax is officially not supported for hidden elements but if we
+        // specify event manually it works.
+        '#ajax' => [
+          'callback' => [get_class($this), 'updateWidgetCallback'],
+          'wrapper' => $details_id,
+          'event' => 'entity_browser_selected_updated',
+        ],
+      ],
+    ];
+
+    $form['current'] = [
+      '#theme_wrappers' => ['container'],
+      '#attributes' => ['class' => ['selected-entities-list']],
+      'items' => array_map(
+        function($id) use ($selected_entities, $details_id) {
+          $entity = $selected_entities[$id];
+          return [
+            '#theme_wrappers' => ['container'],
```

Comment:
We will need to add some kind of hidden element for weight, which will be updated by JS when reordering.

Then we will need to check those values in submit and make sure entities are reordered according to those weights.